### PR TITLE
Placed help button location to after question title

### LINKED ISSUE #1965: Move help button location to be after question title
**Problem**
Current help button is located after instructions field, which looks awkward if there's no instructions text

**Describe the solution you'd like**
EmCare implementation moved the help button to be located after the question title. 

**Describe alternatives you've considered**
Could make this configurable, but don't think it is necessary at this time.

**Additional context**
EmCare commit https://github.com/google/android-fhir/commit/c130243616a9c355e8e3b5a9f25673e817769336

**Screenshot**
Screenshot 2023-04-12 at 14 47 25

cc @santosh-pingle

### PR DESCRIPTION
**Pull request by khyativyasargus**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #1965 

**Description**
Move help button location to after question title as subtitle might be empty in some cases.

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: Feature

**Screenshots (if applicable)**
image

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

**Comment by santosh-pingle** [comment]
@khyativyasargus 
I noticed a slight cut on the right side of the help button. Could you please confirm this?

Can you also try the following precondition and see how the help button aligns?
Please modify questionnaire.json and add 1. precondition text 2. long question text

**Comment by khyativyasargus** [comment]
Thanks @santosh-pingle for drawing my attention. Sure will check and provide requested details. Just one doubt can you elaborate about precondition here?

**Comment by santosh-pingle** [comment]
> Thanks @santosh-pingle for drawing my attention. Sure will check and provide requested details. Just one doubt can you elaborate about precondition here?

Actually, i just wanted to see if prefix text is present and question text is 2-3 line text in the json file then how help button get aligns`.

In the component_help.json, you can add `"prefix":"3."` and `"text": "Adding long text to just see how help button get aligns, its just testing.Adding long text to just see how help button get aligns, its just testing.Adding long text to just see how help button get aligns, its just testing",`

**Comment by khyativyasargus** [comment]
> I noticed a slight cut on the right side of the help button. Could you please confirm this?

@santosh-pingle  This is because for help button, Button widget is used which is ideally in square shape with edges and we are putting round shape image into that.
so to solve that there are 2 options in my mind right now,
1)increase button width and height by 1dp and put 1dp padding 
2)change widget type from Button to ImageView 

what do you suggest?  Let me know if you have any other option on your mind , I'll do changes accordingly.

**Comment by khyativyasargus** [comment]
> Actually, i just wanted to see if prefix text is present and question text is 2-3 line text in the json file then how help button get aligns`.
> 
> In the component_help.json, you can add `"prefix":"3."` and `"text": "Adding long text to just see how help button get aligns, its just testing.Adding long text to just see how help button get aligns, its just testing.Adding long text to just see how help button get aligns, its just testing",`

@santosh-pingle I've done some corrections to manage long text, but I'll commit them with help button changes together. Here it is SS for how it looks

image

**Comment by santosh-pingle** [comment]
> > Actually, i just wanted to see if prefix text is present and question text is 2-3 line text in the json file then how help button get aligns`.
> > In the component_help.json, you can add `"prefix":"3."` and `"text": "Adding long text to just see how help button get aligns, its just testing.Adding long text to just see how help button get aligns, its just testing.Adding long text to just see how help button get aligns, its just testing",`
> 
> @santosh-pingle I've done some corrections to manage long text, but I'll commit them with help button changes together. Here it is SS for how it looks
> 
> image
> 
> image
> 
> image

@shelaghm Can you please review attached screenshots by @khyativyasargus 

You mean, changes will be updated to this pr, correct?

**Comment by khyativyasargus** [comment]
> @khyativyasargus, you mean, changes will be updated to this pr, correct?

Yes @santosh-pingle , once you & @shelaghm approves screenshots I'll commit changes in this PR only.

**Comment by khyativyasargus** [comment]
> @khyativyasargus thanks for this change, would it be possible to also refactor the header so that the prefix is also inline the same way the icon is inline in the link i posted above?

Yes @jingtang10 this option sounds more convenient to me as per UI perspective. so you wan prefix & help icon both inline with title right?  I'll give it check and will update you with changes.

**Comment by khyativyasargus** [comment]
> @khyativyasargus thanks for this change, would it be possible to also refactor the header so that the prefix is also inline the same way the icon is inline in the link i posted above?

Hi @jingtang10 @santosh-pingle ! As per our conversation, I've done with refactoring the question header view in such a way that the Prefix, Title, and icon stay inline with the help of Spannable. 
But I think Text & Icon should also be customized with custom style attributes and I am not sure how we can achieve that. Can you guide me about the possibility?

**Comment by jingtang10** [comment]
hey @khyativyasargus thanks for the update. can you add new screenshots, especially for long titles?

**Comment by khyativyasargus** [comment]
> hey @khyativyasargus thanks for the update. can you add new screenshots, especially for long titles?

Sure @jingtang10 , Here is a screenshot of inline headerview with longer title

image

**Comment by shelaghm** [comment]
Thanks @khyativyasargus Looks better, one small adjustment would be to make the help button aligned to the text line height? See screenshot as an example Screenshot 2023-05-17 at 08 04 50

File: datacapture/src/main/java/com/google/android/fhir/datacapture/extensions/MoreHeaderViews.kt [hunk 1]
```diff
@@ -34,6 +47,66 @@ internal fun TextView.updateTextAndVisibility(localizedText: Spanned? = null) {
...
+      text.length + 1,
+      Spanned.SPAN_EXCLUSIVE_EXCLUSIVE
+    )
+  }.let {
+    text = it
+    movementMethod = LinkMovementMethod.getInstance()
+  }
+
+  helpTextView.updateTextAndVisibility(questionnaireItem.localizedHelpSpanned)
+}
```
Comment (aditya-07) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/extensions/MoreHeaderViews.kt:104]: Can we remove this code if it is not being used anywhere?

**Review by aditya-07 — CHANGES_REQUESTED** [review]
If we are not depending on any kotlin code to place the Help Button, then we should remove it from this PR.

**Comment by aditya-07** [comment]
@khyativyasargus Please update the PR to latest changes.

**Comment by khyativyasargus** [comment]
> If we are not depending on any kotlin code to place the Help Button, then we should remove it from this PR.

Hi @aditya-07 these changes are required but as we haven't been able to find a way to configure Image using style attribute with Spannable so it's still pending. We can refactor the code of this PR once we find the final solution.

**Comment by jingtang10** [comment]
Given the status of this PR I'm going to close this for now until we have a proper solution that works well. @khyativyasargus if this becomes blocking please feel free to reopen.

**Pull request closed without merging**